### ISSUE TITLE
Shrink snapshot for use with smaller instance?

### ISSUE THREAD

**Issue by AEI-SXSmirror** [Unicamp-OpenPower/minicloud#16]
Not a bug, but a nuisance:
I had started a "tiny" instance to run my tests, but found that one particular task would require more than 3 GB of virtual memory, and even adding a swapfile did not help - 1 GB was too tiny for this one.
So I stopped the instance, took a snapshot, and fired up a new large instance based on that. The big task finished happily.
Now I'd like to go back to l*e*ss resource usage, but don't want to lose my setup. Since the root fs has been grown to only fit a large machine, this appears to be impossible - is there a means to shrink a snapshot to retrofit it on a tiny/smaller VM? I hate to waste resources.

Addition: I just discovered that this instance is using a tiny root fs, not a 80 GB one:
```
root@large:~# df /
Filesystem     1K-blocks    Used Available Use% Mounted on
/dev/vda2       18505996 7257068  10285808  42% /
```

**Comment by AEI-SteffenG** [comment]
I found that the size of the snapshot is somewhat bigger than the root fs (is swap also snapped? the difference is quite close to 2 GB), but that should have no effect on the problem itself.
I've also developed the habit of deleting cached files (/var/cache/pbuilder/aptcache and friends), and fill up the empty disk space with zero bytes temporarily before shutting down the instance - this saves several GBs.
